compiler/rustc_attr_parsing/src/attributes/crate_level.rs

```diff
@@ -94,3 +94,29 @@ impl<S: Stage> SingleAttributeParser<S> for RecursionLimitParser {
         })
     }
 }
+
+pub(crate) struct MoveSizeLimitParser;
+
+impl<S: Stage> SingleAttributeParser<S> for MoveSizeLimitParser {
+    const PATH: &[Symbol] = &[sym::move_size_limit];
+    const ATTRIBUTE_ORDER: AttributeOrder = AttributeOrder::KeepOutermost;
+    const ON_DUPLICATE: OnDuplicate<S> = OnDuplicate::WarnButFutureError;
```

Comment:
Not sure if you want to do that in this PR but `#![move_size_limit]` is still unstable under feature `large_assignments` (RUST-83518), so theoretically we could just bump this to `OnDuplicate::Error`.